.github/workflows/check_links.yml

```diff
@@ -31,14 +31,42 @@ jobs:
           key: cache-lychee-${{ github.sha }}
           restore-keys: cache-lychee-
 
-      # Extract and check URLs directly from token files
+      # Extract and check URLs directly from token files (checking dumped file did not work)
       - name: Check token art URLs
         uses: lycheeverse/lychee-action@v1
         with:
-          args: '${{env.lychee_args}} -- tokens.xml challenge_tokens.xml'
+          args: '${{env.lychee_args}} -- tokens.xml'
           fail: true
           jobSummary: true
 
+      # Extract picture URLs from tokens.xml
+      - name: Extract URLs
+        id: tokens_pic_urls
+        uses: lycheeverse/lychee-action@v1
+        with:
+          args: '--dump --exclude www.w3.org -- tokens.xml'
+          output: lychee/out.md
+          fail: true
+          jobSummary: false
+
+      # Analyse extracted links (1/2)
+      - name: List duplicated image links
+        if: steps.tokens_pic_urls.outcome == 'success'
+        shell: bash
+        # Remove blank lines | trim trailing integers (Scryfall) | sort | count and list duplicates
+        run: |
+          echo "🪞 **Duplicated Image Links**" >> $GITHUB_STEP_SUMMARY
+          grep . lychee/out.md | sed 's/\.jpg?.*/.jpg/' | sort | uniq -cd >> $GITHUB_STEP_SUMMARY
+
+      # Analyse extracted links (2/2)
+      - name: List image hosting sources
+        if: steps.tokens_pic_urls.outcome == 'success'
+        shell: bash
+        # Extract domains from URLs | remove blank lines | sort | count and list | sort descending
+        run: |
+          echo "📊 **Image Hosting Statistics**" >> $GITHUB_STEP_SUMMARY
+          awk -F/ '{print $3}' lychee/out.md | grep . | sort | uniq -c | sort -nr >> $GITHUB_STEP_SUMMARY
```

Comment:
why is grep . here? is there empty lines? why does it run sort twice?

Reply:
>why is grep . here? is there empty lines? 

There used to be empty lines in the output when we had token or sets added without image links. This is currently not the case.
Since we do not validate for empty/missing links that could happen again.

But I can also remove it.
Am actually not sure if empty lines would even mess with the result or not.

<br>

>why does it run sort twice?

You mean sorting descending directly before count + list?
I did simply not think about combining that.

Reply:
I guess the comment on it could be more explicit, it takes the third column from the file, removes empty lines, sorts and then counts the duplicates, listing them with their count before the url, then sorts again but based on the number that represents the amount of duplicates descending.

Reply:
Updated the wording of the comment.

The sorting needs to be split because first it's sorting to bring same domain names next to reach other for counting to work. Only after combining the entries and adding quantity numbers it can be sorted descending.